bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/AccountServlet.java

```diff
@@ -557,7 +556,7 @@ private void renderAmazonMusicPlaylistIdChannel(Connection connection, Device de
         }
 
         if (playLists != null) {
-            Map<String, @Nullable PlayList @Nullable []> playlistMap = playLists.playlists;
+            Map<String, PlayList @Nullable []> playlistMap = playLists.playlists;
```

Comment:
Did you check that the array can be null but all array contents are non-null? I would expect that the array is nonnull but it contents are nullable, so `@Nullable Playlist[]` would be correct.

Reply:
They are parsed from JSON with Gson so I would expect the array map value to be either non null or nullable with non null array elements. I would think it's using the correct [syntax](https://help.eclipse.org/2020-09/index.jsp?topic=%2Forg.eclipse.jdt.doc.user%2Ftasks%2Ftask-using_null_type_annotations.htm&cp%3D1_3_9_1_3_0&anchor=compatibility_syntax)?

```java
    // annotated array type:
    Object @NonNull[] o2;
    o2 = null;          // NOT OK
    o2 = new Object[1];
    o2[0] = null;       // OK
```

Reply:
Mh. Good point. Then the annotation was wrong before (and not work-around related (because the members of the array are not what determines the result of `Map.get` but the array itself.